examples/platforms/simulation/simul_utils.c

```diff
@@ -90,43 +206,156 @@ void utilsInitSocket(utilsSocket *aSocket, uint16_t aPortBase)
 
     aSocket->mTxFd = fd;
 
-    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-    // Prepare `mRxFd`
+exit:
+    if (aSocket->mTxFd == -1)
+    {
+        exit(EXIT_FAILURE);
+    }
+}
 
-    fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
-    otEXPECT_ACTION(fd != -1, perror("socket(RxFd)"));
+static bool tryInitSocketIfname(utilsSocket *aSocket, const char *aLocalHost)
+{
+    const struct in6_addr *addr6   = NULL;
+    const struct in6_addr *addr6l  = NULL;
+    const struct in_addr  *addr4   = NULL;
+    struct ifaddrs        *ifaddr  = NULL;
+    unsigned int           ifIndex = 0;
 
-    rval = setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &one, sizeof(one));
-    otEXPECT_ACTION(rval != -1, perror("setsockopt(RxFd, SO_REUSEADDR)"));
+    otEXPECT((ifIndex = if_nametoindex(aLocalHost)));
 
-    rval = setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, &one, sizeof(one));
-    otEXPECT_ACTION(rval != -1, perror("setsockopt(RxFd, SO_REUSEPORT)"));
+    if (getifaddrs(&ifaddr) == -1)
+    {
+        perror("getifaddrs");
+        exit(EXIT_FAILURE);
+    }
 
-    memset(&mreq, 0, sizeof(mreq));
-    inet_pton(AF_INET, UTILS_SOCKET_GROUP_ADDR, &mreq.imr_multiaddr);
+    for (struct ifaddrs *ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next)
+    {
+        if (ifa->ifa_addr == NULL || strcmp(ifa->ifa_name, aLocalHost) != 0)
```

Comment:
`ifa_name` seems to be the interface name. https://man7.org/linux/man-pages/man3/getifaddrs.3.html
 
It's comparing the interface name with an IP address. Is this expected?

Reply:
`aLocalHost` can be a interface name.

Reply:
Can we rename the parameter `aLocalHost` and the global variable `gLocalHost`?

Reply:
I'm fine with renaming it in a separate PR.